### PR TITLE
feat(sidebar): add support to create db from connection COMPASS-7885

### PR DESCRIPTION
**Pull request by mabaasit**
https://github.com/mongodb-js/compass/assets/1305718/d1769520-2f44-4387-b4f4-d3ed5e24a12e

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-components/src/components/item-action-controls.tsx [hunk 1]
```diff
@@ -388,6 +392,38 @@ export function ItemActionControls<Action extends string>({
...
+          onAction={onAction}
+          className={cx('item-action-controls', className)}
+          menuClassName={menuClassName}
+          iconClassName={iconClassName}
+          iconStyle={iconStyle}
+          iconSize={iconSize}
+          usePortal={usePortal}
+          data-testid={dataTestId}
+        ></ItemActionMenu>
+      </div>
```
Comment (paula-stacho) [inline packages/compass-components/src/components/item-action-controls.tsx:423]: looks like we're repeating most of these props 4 times, perhaps we can set define them once and reuse

File: packages/compass-components/src/components/item-action-controls.tsx [hunk 2]
```diff
@@ -10,14 +10,14 @@ import {
...
-
+import type { glyphs } from '@leafygreen-ui/icon';
 import { spacing } from '@leafygreen-ui/tokens';
 import { css, cx } from '@leafygreen-ui/emotion';
 
 export type ItemAction<Action extends string> = {
   action: Action;
   label: string;
-  icon: React.ReactChild;
+  icon: keyof typeof glyphs | React.ReactElement;
```
Comment (paula-stacho) [inline packages/compass-components/src/components/item-action-controls.tsx:20]: oh wow, good find!

**Review by himanshusinghs — APPROVED** [review]
🚀

**Comment by himanshusinghs** [comment]
image
Maybe a small thing - we probably don't need the title when the tooltip is open?

**Pull request merged**